tests/unit/planning/test_layer_planner.py

```
@@ -0,0 +1,288 @@
+    # k <= 1 returns before the policy is ever inspected -- even an unknown one.
+    out = assign_to_layers(four_drones, FakeLayerStack([500.0]), policy)
+    assert _ids(out) == {0: [0, 1, 2, 3]}
+
+
+def test_single_policy_puts_everyone_on_layer_zero(four_drones):
+    out = assign_to_layers(four_drones, FakeLayerStack([100.0, 100.0, 100.0]), POLICY_SINGLE)
+    assert _ids(out) == {0: [0, 1, 2, 3]}
+
+
+def test_unknown_policy_raises_value_error(four_drones):
+    with pytest.raises(ValueError) as exc:
```

Ruff (0.16.3)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)